services/checks/checks.go

```diff
@@ -470,22 +491,29 @@ func (s *Service) executeMongoDBChecks(ctx context.Context) []string {
 				continue
 			}
 
-			alerts, err := s.processResults(ctx, c, target, r.ID)
+			checkResults, err := s.processResults(ctx, c, target, r.ID)
 			if err != nil {
 				s.l.Warnf("Failed to process action result: %s", err)
 				continue
 			}
 
 			s.mScriptsExecuted.WithLabelValues(string(models.MongoDBServiceType)).Inc()
-			s.mAlertsGenerated.WithLabelValues(string(models.MongoDBServiceType), string(c.Type)).Add(float64(len(alerts)))
-			res = append(res, alerts...)
+			s.mAlertsGenerated.WithLabelValues(string(models.MongoDBServiceType), string(c.Type)).Add(float64(len(checkResults)))
+			res = append(res, checkResults...)
 		}
 	}
 
 	return res
 }
 
-func (s *Service) processResults(ctx context.Context, check check.Check, target target, resID string) ([]string, error) {
+type sttCheckResult struct {
+	checkName string
+	target    target
+	result    check.Result
+	alertID   string
```

Comment:
It should not be there

Reply:
but I need alertID later on to pass in the `createAlert` call

Reply:
Yes. What I'm saying is that one point of our refactoring is to decouple check results from alerts. Therefore, `alertID` does not belong to `sttCheckResult`. I think it can be computed later?

Reply:
got it